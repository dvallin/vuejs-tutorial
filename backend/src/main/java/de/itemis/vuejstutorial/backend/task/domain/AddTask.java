package de.itemis.vuejstutorial.backend.task.domain;

import com.fasterxml.jackson.annotation.JsonProperty;

import javax.validation.constraints.NotBlank;

public class AddTask {
    @NotBlank(message = "Task title must not be blank!")
    public final String title;

    public AddTask(@JsonProperty("title") final String title) {
        this.title = title;
    }
}
