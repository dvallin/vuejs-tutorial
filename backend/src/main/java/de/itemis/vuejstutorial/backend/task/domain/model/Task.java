package de.itemis.vuejstutorial.backend.task.domain.model;

import org.springframework.data.annotation.Id;

import javax.validation.constraints.NotBlank;
import java.util.UUID;

public class Task {
    @Id
    public final UUID id;
    
    public final String title;

    public Task(final String title) {
        this.id = UUID.randomUUID();
        this.title = title;
    }
}
