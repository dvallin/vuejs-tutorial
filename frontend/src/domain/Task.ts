export enum TaskState {
  Open,
  Done
}

export class Task {
  title: string;
  state: TaskState;

  constructor(title: string) {
    this.title = title;
    this.state = TaskState.Open;
  }
}
