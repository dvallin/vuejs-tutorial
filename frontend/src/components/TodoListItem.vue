<template>
  <div class="todo-list-item">
    <input type="checkbox" v-model="isComplete"/>
    <label>{{task.title}}</label>
  </div>
</template>

<script lang="ts">
  import {TaskState} from "../domain/Task";
  import {Actions} from "../store/actions";

  export default {
    name: 'TodoListItem',
    props: ['task'],

    data() {
      return { isComplete: this.task.state === TaskState.Done }
    },
    watch: {
      isComplete: 'handleCompleteClick'
    },
    methods: {
      handleCompleteClick() {
        this.$store.dispatch({
          type: Actions.SetTaskComplete,
          complete: this.isComplete,
          title: this.task.title
        });
      }
    }
  }
</script>
