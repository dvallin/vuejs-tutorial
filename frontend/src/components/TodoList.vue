<template>
  <b-list-group>
    <b-list-group-item class="task" v-for="task in tasks" :key="task.name">
      <todo-list-item :task="task"></todo-list-item>
    </b-list-group-item>
  </b-list-group>
</template>

<script lang="ts">
  import TodoListItem from './TodoListItem'

  export default {
    name: 'TodoList',
    props: ['tasks'],
    components: {
      TodoListItem
    }
  }
</script>
