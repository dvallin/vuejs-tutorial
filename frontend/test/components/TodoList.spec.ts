import TodoList from '../../src/components/TodoList'
import TodoListItem from '../../src/components/TodoListItem'
import {Task} from "../../src/domain/Task";
import {shallow} from 'vue-test-utils'

const task = new Task("id", "Some Title");

describe('TodoList.vue', () => {
  it('renders empty lists', () => {
    const cmp = shallow(TodoList,  {
      propsData: {
        tasks: []
      }
    });
    expect(cmp.element).toMatchSnapshot();
    expect(cmp.vm.$el.childNodes.length).toBe(0);
  });

  it('renders three element lists', () => {
    const cmp = shallow(TodoList,  {
      propsData: {
        tasks: [task,task,task]
      }
    });
    expect(cmp.element).toMatchSnapshot();
    expect(cmp.find(TodoListItem).vm.task.title).toEqual('Some Title');
    expect(cmp.findAll(TodoListItem).length).toBe(3);
  });
});
