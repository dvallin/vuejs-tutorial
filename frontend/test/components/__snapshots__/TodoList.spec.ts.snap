// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList.vue renders empty lists 1`] = `
<ul
  class="todo-list"
/>
`;

exports[`TodoList.vue renders three element lists 1`] = `
<ul
  class="todo-list"
>
  <li
    class="task"
  >
    <!---->
  </li>
  <li
    class="task"
  >
    <!---->
  </li>
  <li
    class="task"
  >
    <!---->
  </li>
</ul>
`;
